EnMasseProject/enmasse
# Go 1.12 and updated go types

Fixes https://github.com/EnMasseProject/enmasse/issues/2942

# Motivation
The AeroGear team is using enmasse/AMQ Online in an operator and our operator is written in go.  This PR updates the address object to have the Status field.

#How to test
Write a simple operator, go dep add the branch the PR is coming from, fetch an address and make sure its status matches the status of the object in k8s.

I couldn't find any integration tests in the project, but I did test the code locally with my operator.
